src/matvis/cpu.py

```diff
@@ -264,13 +270,21 @@ def simulate(
     max_progress_reports : int, optional
         Maximum number of progress reports to print to the screen (if logging level
         allows). Default is 100.
+    use_loop_over_antpairs : bool, optional
+        Whether to calculate visibilities for each antpair in antpairs as a vector
+        dot-product instead of using a full matrix-matrix multiplication for all
+        possible pairs. Default is False. Setting to True can be faster for large
+        arrays where `antpairs` is small (possibly from high redundancy). You should
+        run a performance test before using this.
 
     Returns
     -------
     vis : array_like
         Simulated visibilities. If `polarized = True`, the output will have
         shape (NTIMES, NFEED, NFEED, NANTS, NANTS), otherwise it will have
-        shape (NTIMES, NANTS, NANTS).
+        shape (NTIMES, NANTS, NANTS). If ``antpairs`` is provided, the shape will have
```

Comment:
In the declaration of the vis array below, it looks like the shape should be (NTIMES,NFEED*NANTS,NFEED*NANTS) for the full matrix multiplication case.